registry/servicediscovery/store/cache_manager_test.go

```diff
@@ -27,7 +27,7 @@ import (
 )
 
 func TestCacheManager(t *testing.T) {
-	cm, err := NewCacheManager("test", "test_cache", time.Second, 10)
+	cm, err := NewCacheManager("test", "test_cache", time.Second*3, 10)
```

Comment:
Why does the time interval affect the results? Does 3 seconds guarantee these tests pass?

Reply:
I dont think it would pass every time, so I update the logic to make surethat the next cm created will be able to read the cache file